tests/Unit/Validators/UpdateApplicationProfileValidatorTest.php

```diff
@@ -0,0 +1,173 @@
+            'tagline' => 'GO FOR IT',
+        ];
+        $this->be($applicant->user);
+        $validator = new UpdateApplicationProfileValidator($applicant);
+        $this->assertTrue($validator->isValid($data));
+    }
+
+    public function testUpdatePasswordFailsWithoutConfirm()
+    {
+        $user = factory(User::class)->create([
+            'password' => Hash::make('Testing123!')
+        ]);
+        $applicant = factory(Applicant::class)->create(['user_id' => $user->id]);
+        $data = [
+            'profile_name' => $applicant->user->name,
+            'profile_email' => $applicant->user->email,
+            'old_password' => 'Testing123!',
+            'new_password' => 'NewPassword123!',
+            'twitter_username' => 'Test_person',
+            'linkedin_url' => 'www.linkedin.com/in/test-person',
+            'tagline' => 'GO FOR IT',
+        ];
+        $this->be($applicant->user);
+        $validator = new UpdateApplicationProfileValidator($applicant);
+        $this->assertFalse($validator->isValid($data));
+    }
+
+    public function testUpdatePasswordFailsWithBadConfirm()
+    {
+        $user = factory(User::class)->create([
+            'password' => Hash::make('Testing123!')
+        ]);
+        $applicant = factory(Applicant::class)->create(['user_id' => $user->id]);
+        $data = [
+            'profile_name' => $applicant->user->name,
+            'profile_email' => $applicant->user->email,
+            'old_password' => 'Testing123!',
+            'new_password' => 'NewPassword123!',
+            'new_password_confirmation' => 'DifferentPassword123!',
+            'twitter_username' => 'Test_person',
+            'linkedin_url' => 'www.linkedin.com/in/test-person',
+            'tagline' => 'GO FOR IT',
+        ];
+        $this->be($applicant->user);
+        $validator = new UpdateApplicationProfileValidator($applicant);
+        $this->assertFalse($validator->isValid($data));
+    }
+
+    public function testUpdatePasswordFailsWithIllegalPassword()
+    {
+        $user = factory(User::class)->create([
+            'password' => Hash::make('Testing123!')
+        ]);
+        $applicant = factory(Applicant::class)->create(['user_id' => $user->id]);
+        $data = [
+            'profile_name' => $applicant->user->name,
+            'profile_email' => $applicant->user->email,
+            'old_password' => 'Testing123!',
+            'new_password' => 'NewPassword',
+            'new_password_confirmation' => 'NewPassword',
+            'twitter_username' => 'Test_person',
+            'linkedin_url' => 'www.linkedin.com/in/test-person',
+            'tagline' => 'GO FOR IT',
+        ];
+        $this->be($applicant->user);
+        $validator = new UpdateApplicationProfileValidator($applicant);
+        $this->assertFalse($validator->isValid($data));
+    }
+
+    public function testUpdatePasswordFailsWithBadOldPassword()
+    {
+        $applicant = factory(Applicant::class)->create();
```

Comment:
I _think_ an error was thrown because the factory doesn't use unique email addresses (necessarily). Should be able to resolve this by updating the factory with `->unique()` on the email address.

You can also add a `setUp()` method and create a couple applicants there instead of a new one in every test, saves a bit of time accessing the DB. See the JobControllerTest for an example.

Reply:
Hmm my bad, it's set to unique already. Not sure why it's erroring.